File: 206.py
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        pre = head       # 申请节点 / Request node
        cur = None
        while(pre != None):
            t = pre.next   # 记录下一个节点 / Record next node
            pre.next = cur  # 指到前一个节点 / Refers to the previous node
            cur = pre   # 两个节点都前进一位 / Both nodes move forward
            pre = t
        return cur

File: 206.java
class Solution {
   public ListNode reverseList(ListNode head) {
        ListNode pre = head;  //  request node
        ListNode cur = null;
        ListNode t;
        while(pre != null){
            t = pre.next;   // Record next node
            pre.next = cur;  // Refers to the previous node
            cur = pre;   // 两个节点都前进一位 / Both nodes move forward
            pre = t;
        }
        return cur;
    }
}
